index.bs

```diff
@@ -4445,6 +4465,139 @@ Of the extended attributes defined in this specification, only the [{{Exposed}}]
+as a logical unit.
+
+The <dfn>module specifier</dfn> of a [=module=] is the
+<a href="#prod-string">string</a> immediately following <emu-t>module</emu-t>.
+
+A module is a specification of a set of
+<dfn id="dfn-module-member" export lt="module member">module members</dfn>
+(matching <emu-nt><a href="#prod-ModuleMembers">ModuleMembers</a></emu-nt>),
+which are the [=interfaces=], [=regular operations=] and [=read only=] [=regular
+attributes=] that appear between the braces in the module declaration. These
+operations and attributes describe the behaviors packaged into the module.
+
+The IDL for modules can be split into multiple parts by using <dfn
+id="dfn-partial-module" export>partial module</dfn> definitions (matching
+<emu-t>partial</emu-t> <emu-nt><a href="#prod-Module">Module</a></emu-nt>).
+The module specifier of a partial module definition must be the same as the
+identifier of a module definition. All of the members that appear on each of
+the partial module definitions are considered to be members of the module
+itself.
+
+<pre class="syntax">
+    module <mark>"std:example"</mark> {
+      /* module_members... */
+    };
+
+    partial module <mark>"std:example"</mark> {
+      /* module_members... */
+    };
+</pre>
+
+Note: As with partial interface definitions, partial module definitions are
+intended for use as a specification editorial aide, allowing the definition
+of a module to be separated over more than one section of the document, and
+sometimes multiple documents.
+
+The order that members appear in has significance for property enumeration in
+the <a href="#es-modules">ECMAScript binding</a>.
+
+Note that like namespaces, modules do not create types.
+
+Of the extended attributes defined in this specification, only the [{{Exposed}}]
+and [{{SecureContext}}] extended attributes are applicable to modules. No
+extended attributes are applicable to [=partial modules=].
+
+[=Modules=] must be annotated with the [{{Exposed}}] [=extended attribute=].
+
+[=Module members=], [=partial modules=], and [=interface members=] of
+[=interfaces=] and [=partial interfaces=] which are [=module members=], must not
+be annotated with the [{{Exposed}}] or [{{SecureContext}}] extended attributes.
+
+Note: This restriction could be relaxed if there is a sufficient use case;
+it is omitted here for simplicity of specification and implementation, and a
+lack of clear use cases. Please <a
+href="https://github.com/heycam/webidl/issues/new?title=Fine%20grained%20exposure%20in%20modules">
+file an issue</a> if you see a need for this feature.
+
+Each module may have <dfn>module evaluation steps</dfn> which are executed
+when the module is first imported. These steps may throw an [=exception=], which
+prevents the module from being accessed.
+
+Rather than defining behavior when accessed, the attributes of a [=module=] act
+as a simple storage location. The steps to "set a module attribute" in a
```

Comment:
xref "set a module attribute"?

Reply:
I didn't reference this since that would reference the ECMAScript version; each language binding is responsible for defining its own algorithm.